crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -223,85 +226,40 @@ impl SystemStage {
-            Err(DuplicateLabel(label)) => {
-                panic!("Label {:?} already used by a parallel system.", label)
-            }
-            Err(GraphCycles(labels)) => {
-                panic!(
-                    "Found a dependency cycle in parallel systems: {:?}.",
-                    labels
-                )
-            }
-        }
-        match sort_systems(&mut self.exclusive_at_start) {
-            Ok(()) => (),
-            Err(LabelNotFound(label)) => {
-                panic!(
-                    "No exclusive system with label {:?} at start of stage.",
-                    label
-                )
-            }
-            Err(DuplicateLabel(label)) => {
-                panic!(
-                    "Label {:?} already used by an exclusive system at start of stage.",
-                    label
-                )
-            }
-            Err(GraphCycles(labels)) => {
-                panic!(
-                    "Found a dependency cycle in exclusive systems at start of stage: {:?}.",
-                    labels
-                )
-            }
-        }
-        match sort_systems(&mut self.exclusive_before_commands) {
-            Ok(()) => (),
-            Err(LabelNotFound(label)) => {
-                panic!(
-                    "No exclusive system with label {:?} before commands of stage.",
-                    label
-                )
-            }
-            Err(DuplicateLabel(label)) => {
-                panic!(
-                    "Label {:?} already used by an exclusive system before commands of stage.",
-                    label
-                )
-            }
-            Err(GraphCycles(labels)) => {
-                panic!(
-                    "Found a dependency cycle in exclusive systems before commands of stage: {:?}.",
-                    labels
-                )
-            }
-        }
-        match sort_systems(&mut self.exclusive_at_end) {
-            Ok(()) => (),
-            Err(LabelNotFound(label)) => {
-                panic!(
-                    "No exclusive system with label {:?} at end of stage.",
-                    label
-                )
-            }
-            Err(DuplicateLabel(label)) => {
-                panic!(
-                    "Label {:?} already used by an exclusive system at end of stage.",
-                    label
-                )
-            }
-            Err(GraphCycles(labels)) => {
-                panic!(
-                    "Found a dependency cycle in exclusive systems at end of stage: {:?}.",
-                    labels
-                )
+        fn sort_systems_unwrap(
+            systems: &mut Vec<impl SystemContainer>,
+            systems_description: &'static str,
+        ) {
+            if let Err(DependencyGraphError::GraphCycles(cycle)) = sort_systems(systems) {
+                use std::fmt::Write;
+                let mut string = format!("Found a dependency cycle in {}:", systems_description);
```

Comment:
It feels like `string` should be unwrapped here, rather than repeatedly below.

Reply:
That's not what's happening? The `unwrap`s are "handling" errors `write!` might return.